scapy/contrib/rpl.py

```diff
@@ -0,0 +1,294 @@
+                   ByteField("ver", 0),
+                   ShortField("rank", 1),
+                   BitField("G", 1, 1),
+                   BitField("unused1", 0, 1),
+                   BitEnumField("mop", 1, 3, rplmop),
+                   BitField("prf", 0, 3),
+                   ByteField("dtsn", 240),
+                   ByteField("flags", 0),
+                   ByteField("reserved", 0),
+                   IP6Field("dodagid", "::1")]
+    overload_fields = {RPL: {"code": 1}}
+
+
+class _OptDODAGIDField(IP6Field):
+    def addfield(self, pkt, s, val):
+        if pkt.D == 1:
+            return s + self.i2m(pkt, val)
+        if val:
+            print("RPL DAO 'D' flag is not set but dodagid is given.")
+        return s
+
+
+class DAO(RPLGuessOption, Packet):
+    name = "Destination Advertisement Object"
+    fields_desc = [XShortField("cksum", None),
+                   # Base Object
+                   ByteField("RPLInstanceID", 50),
+                   BitField("K", 0, 1),
+                   BitField("D", 0, 1),
+                   BitField("flags", 0, 6),
+                   ByteField("reserved", 0),
+                   ByteField("daoseq", 1),
+                   _OptDODAGIDField("dodagid", None)]
+    overload_fields = {RPL: {"code": 2}}
+
+
+class DAOACK(RPLGuessOption, Packet):
+    name = "Destination Advertisement Object Acknowledgement"
+    fields_desc = [XShortField("cksum", None),
+                   # Base Object
+                   ByteField("RPLInstanceID", 50),
+                   BitField("D", 0, 1),
+                   BitField("reserved", 0, 7),
+                   ByteField("daoseq", 1),
+                   ByteField("status", 0),
+                   _OptDODAGIDField("dodagid", None)]
+    overload_fields = {RPL: {"code": 3}}
+
+
+# https://datatracker.ietf.org/doc/draft-ietf-roll-efficient-npdao/
+class DCO(RPLGuessOption, Packet):
+    name = "Destination Cleanup Object"
+    fields_desc = [XShortField("cksum", None),
+                   # Base Object
+                   ByteField("RPLInstanceID", 50),
+                   BitField("K", 0, 1),
+                   BitField("D", 0, 1),
+                   BitField("flags", 0, 6),
+                   ByteField("status", 0),
+                   ByteField("dcoseq", 1),
+                   _OptDODAGIDField("dodagid", None)]
+    overload_fields = {RPL: {"code": 7}}
+
+
+# https://datatracker.ietf.org/doc/draft-ietf-roll-efficient-npdao/
+class DCOACK(RPLGuessOption, Packet):
+    name = "Destination Cleanup Object Acknowledgement"
+    fields_desc = [XShortField("cksum", None),
+                   # Base Object
+                   ByteField("RPLInstanceID", 50),
+                   BitField("D", 0, 1),
+                   BitField("flags", 0, 7),
+                   ByteField("dcoseq", 1),
+                   ByteField("status", 0),
+                   _OptDODAGIDField("dodagid", None)]
+    overload_fields = {RPL: {"code": 8}}
```

Comment:
This can be replaced by a `bind_top_down` call (it's cleaner). I know the user of overload_fields is common in Scapy but it's not something we want to keep.

Reply:
Fixed: Removed `overload_fields` from all over RPL.